### ISSUE TITLE
Accessibility of Users landing page

### ISSUE THREAD

**Issue by ajanec01** [craftcms/cms#7486]
### Overview of accessibility issues found on the Users landing page

- Status updates are not picked up by screen readers-  “User unsuspended”,  "User suspended", “User deleted”
![status-update](https://user-images.githubusercontent.com/47136246/106673059-f8528200-65a8-11eb-8700-d7583126f807.png)
- Dialog for deleting a user would benefit from `role=”dialog` and an accessible name
![remove -selected-user](https://user-images.githubusercontent.com/47136246/106673398-7151d980-65a9-11eb-94ed-e0ef32ad1098.png)
- The same dialog does not trap keyboard focus
- The remove button that deletes selected user is not keyboard accessible
- Label for transferring content to another user might be a bit confusing, it says “transfer to”; my suggestion is to give the radio button the “transfer to another user” label and append the choose user button below the radio buttons
- It would also be good to wrap those radio buttons with `role=”radiogroup”` and associate the legend using `aria-labelledby`
- It’s not possible to select a user that I would like to transfer the content to when using the keyboard only
- Checkboxes in the thumbnail view should have `role=”checkbox”` and `aria-checked`; it’s also a bit confusing that they can be navigated using arrow keys only; I think that it would be better if each of them  was a Tab stop
![checkboxes](https://user-images.githubusercontent.com/47136246/106673582-b7a73880-65a9-11eb-8791-b2e2130aeea1.png)
- Edit user pop-up would benefit from role=”dialog” and an accessible name
![error](https://user-images.githubusercontent.com/47136246/106673700-ee7d4e80-65a9-11eb-89d1-13ec6855f466.png)
- Error message for missing the required field in the edit pop-up not communicated to screen reader users; my suggestion is to move keyboard focus to the incorrect field and associate the error message with the input using `aria-describedby`
- Username field in the edit user popup would benefit from the `required `attribute

### Additional info

- Craft version: 3.6.0.1
